src/__tests__/server/db/actions/GameAction.unit.test.ts

```diff
@@ -0,0 +1,355 @@
+      }
+      //Ensure that the lists of objectids with that tag matches exactly what the tag has in the games array.
+      const gameIds = games.map((game) => game._id.toString());
+      gameIds.sort();
+      theme.games.sort();
+      expect(gameIds).toStrictEqual(theme.games);
+    });
+    test("[name] regex case insensitivity in-exact match: expect success", async () => {
+      //Randomly pick the name of a game from generatedGames
+      //Take a random subslice of that game, with random capitilization
+      //Call the action and expect that game to be in generated Games.
+      const randomGame = faker.helpers.arrayElement(generatedGames);
+      const randomName = randomGame.name;
+      const substring = randomlyCapitalizeSubString(randomName);
+      let results = await getSelectedGames({ name: substring, page: 1 });
+      let games = results.games;
+      const numPages = Math.ceil(results.count / RESULTS_PER_PAGE) - 1; //-1 Since one page has already ben traversed.
+      for (let page = 2; page < 2 + numPages; page++) {
+        const result = await getSelectedGames({
+          name: substring,
+          page: page,
+        });
+        games = [...games, ...result.games];
+      }
+      const selectedNames = games.map((game) => game.name);
+      expect(selectedNames).toContain(randomName);
+    });
+    // ...
+    test("happy: expect success", async () => {
+      const customTags = await TagModel.find({ type: "custom" });
+      const customTagNames = customTags.map((tag) => tag.name);
+      const randomCustomTag =
+        customTagNames[Math.floor(Math.random() * customTagNames.length)];
+
+      const accessibilityTags = await TagModel.find({
+        type: "accessibility",
+      });
+      const accessibilityTagNames = accessibilityTags.map((tag) => tag.name);
+      const randomAccessibilityTag =
+        accessibilityTagNames[
+          Math.floor(Math.random() * accessibilityTagNames.length)
+        ];
+
+      const themes = await ThemeModel.find({});
+      const themeNames = themes.map((theme) => theme.name);
+      const randomTheme =
+        themeNames[Math.floor(Math.random() * themeNames.length)];
+
+      const query = {
+        page: 1,
+        tags: [randomCustomTag],
+        accessibility: [randomAccessibilityTag],
+        theme: randomTheme,
+        name: "a",
+        gameBuilds: [
+          AppType.amazon,
+          AppType.appstore,
+          AppType.mac,
+          AppType.webgl,
+        ],
+        gameContent: [GameContentEnum.parentingGuide],
+      };
+      let actual: {
+        games: ExtendId<IGame>[];
+        count: number;
+      };
+      try {
+        actual = await getSelectedGames(query);
+      } catch (e) {
+        if (e instanceof GameNotFoundException) {
+          actual = { games: [], count: 0 }; //Accounts for the case when we grab none of the games via the test.
+        } else {
+          throw e;
+        }
+      }
```

Comment:
Oops, forgot to leave a comment on this - avoid using a try-catch block in a unit test. You should aim to be use matchers using `expect` whenever possible.